packages/chain-bridge/src/adapters/LayerZeroAdapter.sol

```diff
@@ -170,6 +206,122 @@ contract LayerZeroAdapter is
                         EXTERNAL INTERFACE
     //////////////////////////////////////////////////////////////*/
 
+    /// @inheritdoc IAssetAdapter
+    /**
+     * @dev Transfers assets cross-chain using LayerZero OFT protocol.
+     *      Flow: pull tokens → approve OFT → send via OFT → refund excess native.
+     *      If message is provided, encodes as OFT compose for post-delivery execution.
+     * @param operationId Unique identifier for this transfer operation
+     * @param params Transfer parameters (asset, amount, destination, optional message)
+     */
+    function transferAsset(
+        bytes32 operationId,
+        BridgeTypes.ExecuteTransferParams calldata params,
+        BridgeTypes.BridgeOptions calldata
+    )
+        external
+        payable
+        onlyTrustedDestination(params.destinationChainId)
+        onlyRouter
+        nonReentrant
+    {
+        // 1. Validate and get OFT
+        address oft = _getOFTForAsset(params.asset, params.amount);
+
+        // 2. Pull tokens
+        IERC20(params.asset).safeTransferFrom(
+            msg.sender,
+            address(this),
+            params.amount
+        );
+
+        // 3. Get destination and cache EID
+        uint32 dstEid = _externalIdForChain(params.destinationChainId);
+        address dstAdapter = _getAdapterPeerOrRevert(params.destinationChainId);
+
+        // 4. Approve OFT
+        IERC20(params.asset).forceApprove(oft, params.amount);
+
+        // 5. Create compose message if needed
+        bytes memory composeMsg = params.message.length > 0
+            ? BridgeMessagingHelper.encodeRelayedTransferParams(
```

Comment:
in `StargateAdapter.sol` there is no option to skip the message, or rather - the `RelayedTransferParams` are always encoded and sent. I'm not sure if we should keep this pattern or allow it to be optional as in LZ adapter

Reply:
This is a mistake. We always want composeMsg